webrender/src/internal_types.rs

```diff
@@ -180,6 +182,39 @@ pub enum TextureSampler {
     Geometry,
 }
 
+impl TextureSampler {
+    pub fn color(n: usize) -> TextureSampler {
+        match n {
+            0 => TextureSampler::Color0,
+            1 => TextureSampler::Color1,
+            2 => TextureSampler::Color2,
+            _ => {
+                panic!("There are only 3 color samplers.");
+            }
+        }
+    }
+}
+
+/// Optional textures that can be used as a source in the shaders.
+/// Textures that are not used by the batch are equal to TextureId::invalid().
+#[derive(Copy, Clone, Debug)]
+pub struct PrimitiveBatchTextures {
+    pub colors: [TextureId; 3],
+    pub mask: TextureId,
+}
+
+impl PrimitiveBatchTextures {
+    pub fn no_texture() -> Self {
+        PrimitiveBatchTextures {
+            colors: [TextureId::invalid(); 3],
+            mask: TextureId::invalid(),
+        }
+    }
+}
+
+// In some places we need to temporarily bind a texture to any slot.
+pub const DEFAULT_SAMPLER: TextureSampler = TextureSampler::Color0;
```

Comment:
Perhaps, we can lean more to the "texture" part of the concept, since eventually we'll have separate samplers from textures, and this term (`DEFAULT_SAMPLER`) will be misleading. Maybe `DEFAULT_TEXTURE`?